### ISSUE TITLE
Add third-party relay

### ISSUE THREAD

**Issue by Dough-su** [davidmigloz/pixels2flutter#2]
Is it possible to customize the relay function to help those users who don't have the official api?
For example:
    The official api is: api.openai.com
     The third-party api is a different address, you just need to replace it with another one.

**Comment by davidmigloz** [comment]
Good one, I can do that.

In your case, would the third-party api require some kind of authentication?

**Comment by Dough-su** [comment]
> Good one, I can do that.
> 
> In your case, would the third-party api require some kind of authentication?

It is the same as openai official, the only difference is the url address，The authentication method is the same as the official one, but the transfer station provides their own API key,Currently many transfers are using https://github.com/songquanpeng/one-api/blob/main/README.en.md

**Comment by davidmigloz** [comment]
ah you meant using one-api gotcha. I can add some fields to add custom headers or query params as well in case you want to use a proxy that requires them.

**Comment by Dough-su** [comment]
> ah you meant using one-api gotcha. I can add some fields to add custom headers or query params as well in case you want to use a proxy that requires them.

Yes, there are many Chinese users who cannot directly access the official API of OpenAI. They need additional relays to help them use it. Thank you.

**Comment by davidmigloz** [comment]
Awesome, I'll try to implement it in the following days.
